# Reaccciones Nucleares Binarias 

El proposito de este notebook es el de encontrar todas las reacciones nucleares binarias tal que produzcan ${^{26}_{13}Al}$

$$A+b \rightarrow c + {^{26}_{13}Al}$$.


In [1]:
import pandas as pd
import numpy as np

Importamos una base de datos para encontrar la masa y propiedades de los distintos nucleotidos.
Sacada de https://github.com/ffavela/isonav

In [34]:
datos = pd.read_csv("ATOMIC.MASS.csv", sep=',')
datos

Marcador   N-Z      N      Z      A   EL  \
0          NaN   NaN    NaN    NaN    NaN  NaN   
1          0.0   1.0    1.0    0.0    1.0    n   
2          NaN  -1.0    0.0    1.0    1.0    H   
3          0.0   0.0    1.0    1.0    2.0    H   
4          0.0   1.0    2.0    1.0    3.0    H   
5          NaN  -1.0    1.0    2.0    3.0   He   
6          NaN  -3.0    0.0    3.0    3.0   Li   
7          0.0   2.0    3.0    1.0    4.0    H   
8          NaN   0.0    2.0    2.0    4.0   He   
9          NaN  -2.0    1.0    3.0    4.0   Li   
10         0.0   3.0    4.0    1.0    5.0    H   
11         NaN   1.0    3.0    2.0    5.0   He   
12         NaN  -1.0    2.0    3.0    5.0   Li   
13         NaN  -3.0    1.0    4.0    5.0   Be   
14         0.0   4.0    5.0    1.0    6.0    H   
15         NaN   2.0    4.0    2.0    6.0   He   
16         NaN   0.0    3.0    3.0    6.0   Li   
17         NaN  -2.0    2.0    4.0    6.0   Be   
18         NaN  -4.0    1.0    5.0    6.0    B   
19         0.0   5.0    6.0    1.0    7.0    H   
20         NaN   3.0    5.0    2.0    7.0   He   
21         NaN   1.0    4.0    3.0    7.0   Li   
22         NaN  -1.0    3.0    4.0    7.0   Be   
23         NaN  -3.0    2.0    5.0    7.0    B   
24         0.0   4.0    6.0    2.0    8.0   He   
25         NaN   2.0    5.0    3.0    8.0   Li   
26         NaN   0.0    4.0    4.0    8.0   Be   
27         NaN  -2.0    3.0    5.0    8.0    B   
28         NaN  -4.0    2.0    6.0    8.0    C   
29         0.0   5.0    7.0    2.0    9.0   He   
...        ...   ...    ...    ...    ...  ...   
3324       NaN  59.0  171.0  112.0  283.0   Cn   
3325       NaN  57.0  170.0  113.0  283.0   Ed   
3326       0.0  60.0  172.0  112.0  284.0   Cn   
3327       NaN  58.0  171.0  113.0  284.0   Ed   
3328       0.0  61.0  173.0  112.0  285.0   Cn   
3329       NaN  59.0  172.0  113.0  285.0   Ed   
3330       NaN  57.0  171.0  114.0  285.0   Fl   
3331       0.0  60.0  173.0  113.0  286.0   Ed   
3332       NaN  58.0  172.0  114.0  286.0   Fl   
3333       0.0  61.0  174.0  113.0  287.0   Ed   
3334       NaN  59.0  173.0  114.0  287.0   Fl   
3335       NaN  57.0  172.0  115.0  287.0   Ef   
3336       0.0  60.0  174.0  114.0  288.0   Fl   
3337       NaN  58.0  173.0  115.0  288.0   Ef   
3338       0.0  61.0  175.0  114.0  289.0   Fl   
3339       NaN  59.0  174.0  115.0  289.0   Ef   
3340       NaN  57.0  173.0  116.0  289.0   Lv   
3341       0.0  60.0  175.0  115.0  290.0   Ef   
3342       NaN  58.0  174.0  116.0  290.0   Lv   
3343       0.0  61.0  176.0  115.0  291.0   Ef   
3344       NaN  59.0  175.0  116.0  291.0   Lv   
3345       NaN  57.0  174.0  117.0  291.0   Eh   
3346       0.0  60.0  176.0  116.0  292.0   Lv   
3347       NaN  58.0  175.0  117.0  292.0   Eh   
3348       0.0  61.0  177.0  116.0  293.0   Lv   
3349       NaN  59.0  176.0  117.0  293.0   Eh   
3350       NaN  57.0  175.0  118.0  293.0   Ei   
3351       0.0  60.0  177.0  117.0  294.0   Eh   
3352       NaN  58.0  176.0  118.0  294.0   Ei   
3353       0.0  59.0  177.0  118.0  295.0   Ei   

     O     MASS EXCESS       BINDING ENERGY/A        BETA-DECAY ENERGY  \
0     (keV)                (keV)                    ...                  
1     8071.31714    0.00046      0.0      0.0   B-  ...                  
2     7288.97059    0.00009      0.0      0.0   B-  ...                  
3     13135.72174    0.00011   1112.283    0.000 B- ...                  
4     14949.80611    0.00221   2827.266    0.001 B- ...                  
5     14931.21551    0.00233   2572.681    0.001 B- ...                  
6     -pp   28667#      2000#       -2267#     667# ...                  
7     -n   24621.123    100.000     1720.450   25.00...                  
8     2424.91561    0.00006   7073.915    0.000 B- -...                  
9     -p   25323.186    212.132     1153.761   53.03...                  
10    -nn   32892.440     89.443     1336.360   17.8...                  
11    -n   11231.

In [31]:
DATOS=datos.drop(labels=0)
#Cambiamos el tipo de dato de esta columna para operar con ella (ya que tiene las masas)
DATOS["ATOMIC MASS"]=DATOS["ATOMIC MASS"].astype("int")
DATOS.head()

Marcador  N-Z    N    Z    A  EL  \
1       0.0  1.0  1.0  0.0  1.0   n   
2       NaN -1.0  0.0  1.0  1.0   H   
3       0.0  0.0  1.0  1.0  2.0   H   
4       0.0  1.0  2.0  1.0  3.0   H   
5       NaN -1.0  1.0  2.0  3.0  He   

  O     MASS EXCESS       BINDING ENERGY/A        BETA-DECAY ENERGY  \
1  8071.31714    0.00046      0.0      0.0   B-  ...                  
2  7288.97059    0.00009      0.0      0.0   B-  ...                  
3  13135.72174    0.00011   1112.283    0.000 B- ...                  
4  14949.80611    0.00221   2827.266    0.001 B- ...                  
5  14931.21551    0.00233   2572.681    0.001 B- ...                  

   ATOMIC MASS ATOMIC MASS DECIMALES      V/S  
1         8664               0.91585  0.00049  
2         7825               0.03223  0.00009  
3        14101               0.77812  0.00012  
4        16049               0.27791  0.00237  
5        16029               0.32008   0.0025

Buscamos las propiedades del ${^{26}_{}}Al$

In [6]:
DATOS[DATOS["A"]==26]

Marcador   N-Z     N     Z     A  EL  \
150       0.0  10.0  18.0   8.0  26.0   O   
151       NaN   8.0  17.0   9.0  26.0   F   
152       NaN   6.0  16.0  10.0  26.0  Ne   
153       NaN   4.0  15.0  11.0  26.0  Na   
154       NaN   2.0  14.0  12.0  26.0  Mg   
155       NaN   0.0  13.0  13.0  26.0  Al   
156       NaN  -2.0  12.0  14.0  26.0  Si   
157       NaN  -4.0  11.0  15.0  26.0   P   
158       NaN  -6.0  10.0  16.0  26.0   S   

    O     MASS EXCESS       BINDING ENERGY/A        BETA-DECAY ENERGY  \
150  -nn   34733.037    155.504     6494.709    5.9...                  
151  x   18665.061     77.280     7082.618    2.972...                  
152  x     479.445     18.186     7751.974    0.699...                  
153  x   -6860.780      3.502     8004.201    0.135...                  
154  -16214.546      0.030     8333.870    0.001 B-...                  
155  -12210.112      0.064     8149.764    0.002 B-...                  
156  -   -7140.977      0.107     7924.707    0.004...                  
157  x   10973#       196#        7198#       8#   ...                  
158  x   27079#       596#        6548#      23#   ...                  

     ATOMIC MASS ATOMIC MASS DECIMALES     V/S  
150        37287                  0.45  166.94  
151        20037                 0.768  82.963  
152          514                 0.705  19.523  
153       992634                 0.649   3.759  
154       982592                 0.968   0.031  
155       986891                 0.904   0.069  
156       992333                 0.845   0.114  
157        11780                     #    210#  
158        29070                     #    640#

In [7]:
DATOS.loc[155:155]

Marcador  N-Z     N     Z     A  EL  \
155       NaN  0.0  13.0  13.0  26.0  Al   

    O     MASS EXCESS       BINDING ENERGY/A        BETA-DECAY ENERGY  \
155  -12210.112      0.064     8149.764    0.002 B-...                  

     ATOMIC MASS ATOMIC MASS DECIMALES    V/S  
155       986891                 0.904  0.069

Definimos la función para el calculo de energía.
$$Q=E_f-E_i$$
$$E=mc^2= \mu u \frac{10^{-6} u}{1 \mu u} \frac{931.494 Mev}{1 uc^2}$$

In [8]:
micromassAl26=DATOS["ATOMIC MASS"][155]
def Q(micromassA,micromassb,micromassc):
    Ef=(micromassc+micromassAl26)
    Ei=(micromassA+micromassb)
    MeV=(Ef-Ei)*931.494*1e-06
    return MeV

Busquemos ahora reacciones con nucleos $A,b,c$ que cumplan que su número atomico $A$ sea:

$$A+b \rightarrow c + {^{26}_{13}Al}$$.

Nota: Aquí se define An

In [9]:
Al=26
nAl=float(13)
zAl=float(13)
#Numero de tamaño del nucleo considerado An
An=26
A=[]
for a in range(1,An+1):
    for b in range(1,An+1):
        for c in range(1,An+1):
                if a+b-c==Al:
                    A.append([a,b,c])
print("Así las posibilidades de [a,b,c] donde 26=a+b-c son nucleos con masa atomica hasta de", An, "son:")
len(A)

Así las posibilidades de [a,b,c] donde 26=a+b-c son nucleos con masa atomica hasta de 26 son:


351

Ahora, trabajaremos con la primera posibilidad.

In [10]:
print("Así para la primera posible reacción que produce Al es A+b=c+Al:")
print("A tiene que tener un número atómico de",A[0][0])
print("b tiene que tener un número atómico de",A[0][1])
print("c tiene que tener un número atómico de",A[0][2])

Así para la primera posible reacción que produce Al es A+b=c+Al:
A tiene que tener un número atómico de 1
b tiene que tener un número atómico de 26
c tiene que tener un número atómico de 1


Veremos cuales son los nucleos que cumplen la condición para la conservación de protones y neutrones.
Primero filtramos los datos para que satisfagan la combinación anterior.

In [20]:
Condicion_A0=pd.DataFrame(DATOS[DATOS["A"]==float(A[0][0])])
Condicion_A0

Marcador  N-Z    N    Z    A EL  \
1       0.0  1.0  1.0  0.0  1.0  n   
2       NaN -1.0  0.0  1.0  1.0  H   

  O     MASS EXCESS       BINDING ENERGY/A        BETA-DECAY ENERGY  \
1  8071.31714    0.00046      0.0      0.0   B-  ...                  
2  7288.97059    0.00009      0.0      0.0   B-  ...                  

   ATOMIC MASS ATOMIC MASS DECIMALES      V/S  
1         8664               0.91585  0.00049  
2         7825               0.03223  0.00009

In [21]:
Condicion_b0=DATOS[DATOS["A"]==float(A[0][1])]
Condicion_b0

Marcador   N-Z     N     Z     A  EL  \
150       0.0  10.0  18.0   8.0  26.0   O   
151       NaN   8.0  17.0   9.0  26.0   F   
152       NaN   6.0  16.0  10.0  26.0  Ne   
153       NaN   4.0  15.0  11.0  26.0  Na   
154       NaN   2.0  14.0  12.0  26.0  Mg   
155       NaN   0.0  13.0  13.0  26.0  Al   
156       NaN  -2.0  12.0  14.0  26.0  Si   
157       NaN  -4.0  11.0  15.0  26.0   P   
158       NaN  -6.0  10.0  16.0  26.0   S   

    O     MASS EXCESS       BINDING ENERGY/A        BETA-DECAY ENERGY  \
150  -nn   34733.037    155.504     6494.709    5.9...                  
151  x   18665.061     77.280     7082.618    2.972...                  
152  x     479.445     18.186     7751.974    0.699...                  
153  x   -6860.780      3.502     8004.201    0.135...                  
154  -16214.546      0.030     8333.870    0.001 B-...                  
155  -12210.112      0.064     8149.764    0.002 B-...                  
156  -   -7140.977      0.107     7924.707    0.004...                  
157  x   10973#       196#        7198#       8#   ...                  
158  x   27079#       596#        6548#      23#   ...                  

     ATOMIC MASS ATOMIC MASS DECIMALES     V/S  
150        37287                  0.45  166.94  
151        20037                 0.768  82.963  
152          514                 0.705  19.523  
153       992634                 0.649   3.759  
154       982592                 0.968   0.031  
155       986891                 0.904   0.069  
156       992333                 0.845   0.114  
157        11780                     #    210#  
158        29070                     #    640#

In [22]:
Condicion_c0=DATOS[DATOS["A"]==float(A[0][2])]
Condicion_c0

Marcador  N-Z    N    Z    A EL  \
1       0.0  1.0  1.0  0.0  1.0  n   
2       NaN -1.0  0.0  1.0  1.0  H   

  O     MASS EXCESS       BINDING ENERGY/A        BETA-DECAY ENERGY  \
1  8071.31714    0.00046      0.0      0.0   B-  ...                  
2  7288.97059    0.00009      0.0      0.0   B-  ...                  

   ATOMIC MASS ATOMIC MASS DECIMALES      V/S  
1         8664               0.91585  0.00049  
2         7825               0.03223  0.00009

Añadimos la condición la conservación de n como para z al variar en el dato que buscamos.

In [25]:
for indexA, rowA in Condicion_A0.iterrows():
    for indexb, rowb in Condicion_b0.iterrows():
        for indexc, rowc in Condicion_b0.iterrows():
            if rowA["N"]+rowb["N"]-rowc["N"]==nAl:
                if rowA["Z"]+rowb["Z"]-rowc["Z"]==zAl:
                    print("A index: ",indexA,"C index: ",indexc,"c index: ",indexc)
print("Fin, no hay posibilidades")

Fin, no hay posibilidades


Ahora variamos en todas las posibilidades que nos brindó A, determinamos la reacción y calculamos Q

In [17]:
INDEX=[]
for k in range(0,len(A)):
    Condicion_A=pd.DataFrame(DATOS[DATOS["A"]==float(A[k][0])])
    Condicion_b=DATOS[DATOS["A"]==float(A[k][1])]
    Condicion_c=DATOS[DATOS["A"]==float(A[k][2])]
# Deterinacion que cumpla con el valance de neutrones y protones
    for indexA, rowA in Condicion_A.iterrows():
        for indexb, rowb in Condicion_b.iterrows():
            for indexc, rowc in Condicion_b.iterrows():
                if rowA["N"]+rowb["N"]-rowc["N"]==nAl:
                    if rowA["Z"]+rowb["Z"]-rowc["Z"]==zAl:
                        INDEX.append([indexA,indexb,indexc])
                        print("Reacción:",rowA["A"],rowA["EL"],"+",rowb["A"],rowb["EL"],"=",rowc["A"],rowc["EL"],"+ 26.0 Al")
                        print("Q =: ",Q(rowA["ATOMIC MASS"],rowb["ATOMIC MASS"],rowb["ATOMIC MASS"])," MeV")
                        print("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~")

Reacción: 26.0 Mg + 1.0 H = 1.0 n + 26.0 Al
Q =:  4.004492706  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 Al + 1.0 n = 1.0 n + 26.0 Al
Q =:  0.0  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 Al + 1.0 H = 1.0 H + 26.0 Al
Q =:  0.0  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 Si + 1.0 n = 1.0 H + 26.0 Al
Q =:  -5.069190348  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 Al + 2.0 H = 2.0 H + 26.0 Al
Q =:  0.0  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 Na + 3.0 Li = 3.0 H + 26.0 Al
Q =:  -5.349570042  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 Mg + 3.0 He = 3.0 H + 26.0 Al
Q =:  4.004492706  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 Mg + 3.0 Li = 3.0 He + 26.0 Al
Q =:  4.004492706  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 Al + 3.0 H = 3.0 H + 26.0 Al
Q =:  0.0  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 Al + 3.0 He = 3.0 He + 26.0 Al

Reacción: 26.0 Al + 9.0 C = 9.0 C + 26.0 Al
Q =:  0.0  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 Si + 9.0 He = 9.0 Li + 26.0 Al
Q =:  -5.069190348  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 Si + 9.0 Li = 9.0 Be + 26.0 Al
Q =:  -5.069190348  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 Si + 9.0 Be = 9.0 B + 26.0 Al
Q =:  -5.069190348  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 Si + 9.0 B = 9.0 C + 26.0 Al
Q =:  -5.069190348  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 P + 9.0 He = 9.0 Be + 26.0 Al
Q =:  908.310045834  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 P + 9.0 Li = 9.0 B + 26.0 Al
Q =:  908.310045834  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 P + 9.0 Be = 9.0 C + 26.0 Al
Q =:  908.310045834  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 S + 9.0 He = 9.0 B + 26.0 Al
Q =:  892.204514574  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción:

Reacción: 26.0 Ne + 14.0 O = 14.0 B + 26.0 Al
Q =:  918.804257238  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 Ne + 14.0 F = 14.0 C + 26.0 Al
Q =:  918.804257238  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 Na + 14.0 C = 14.0 Be + 26.0 Al
Q =:  -5.349570042  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 Na + 14.0 N = 14.0 B + 26.0 Al
Q =:  -5.349570042  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 Na + 14.0 O = 14.0 C + 26.0 Al
Q =:  -5.349570042  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 Na + 14.0 F = 14.0 N + 26.0 Al
Q =:  -5.349570042  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 Mg + 14.0 B = 14.0 Be + 26.0 Al
Q =:  4.004492706  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 Mg + 14.0 C = 14.0 B + 26.0 Al
Q =:  4.004492706  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 Mg + 14.0 N = 14.0 C + 26.0 Al
Q =:  4.004492706  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Q =:  900.618699876  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 F + 18.0 Na = 18.0 N + 26.0 Al
Q =:  900.618699876  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 Ne + 18.0 O = 18.0 B + 26.0 Al
Q =:  918.804257238  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 Ne + 18.0 F = 18.0 C + 26.0 Al
Q =:  918.804257238  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 Ne + 18.0 Ne = 18.0 N + 26.0 Al
Q =:  918.804257238  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 Ne + 18.0 Na = 18.0 O + 26.0 Al
Q =:  918.804257238  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 Na + 18.0 N = 18.0 B + 26.0 Al
Q =:  -5.349570042  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 Na + 18.0 O = 18.0 C + 26.0 Al
Q =:  -5.349570042  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 Na + 18.0 F = 18.0 N + 26.0 Al
Q =:  -5.349570042  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 Na + 18.0 Ne = 

Reacción: 26.0 F + 21.0 F = 21.0 B + 26.0 Al
Q =:  900.618699876  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 F + 21.0 Ne = 21.0 C + 26.0 Al
Q =:  900.618699876  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 F + 21.0 Na = 21.0 N + 26.0 Al
Q =:  900.618699876  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 F + 21.0 Mg = 21.0 O + 26.0 Al
Q =:  900.618699876  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 F + 21.0 Al = 21.0 F + 26.0 Al
Q =:  900.618699876  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 Ne + 21.0 O = 21.0 B + 26.0 Al
Q =:  918.804257238  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 Ne + 21.0 F = 21.0 C + 26.0 Al
Q =:  918.804257238  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 Ne + 21.0 Ne = 21.0 N + 26.0 Al
Q =:  918.804257238  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 Ne + 21.0 Na = 21.0 O + 26.0 Al
Q =:  918.804257238  MeV
~~~~~~~~~~~~~~~~~~~~~~~~

Q =:  4.004492706  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 Mg + 23.0 Mg = 23.0 Na + 26.0 Al
Q =:  4.004492706  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 Mg + 23.0 Al = 23.0 Mg + 26.0 Al
Q =:  4.004492706  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 Mg + 23.0 Si = 23.0 Al + 26.0 Al
Q =:  4.004492706  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 Al + 23.0 C = 23.0 C + 26.0 Al
Q =:  0.0  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 Al + 23.0 N = 23.0 N + 26.0 Al
Q =:  0.0  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 Al + 23.0 O = 23.0 O + 26.0 Al
Q =:  0.0  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 Al + 23.0 F = 23.0 F + 26.0 Al
Q =:  0.0  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 Al + 23.0 Ne = 23.0 Ne + 26.0 Al
Q =:  0.0  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 Al + 23.0 Na = 23.0 Na + 26.0 Al
Q =:  0.0  MeV
~~~~~~~~~~~~~~~~

Reacción: 26.0 Si + 24.0 Al = 24.0 Si + 26.0 Al
Q =:  -5.069190348  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 Si + 24.0 Si = 24.0 P + 26.0 Al
Q =:  -5.069190348  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 P + 24.0 N = 24.0 F + 26.0 Al
Q =:  908.310045834  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 P + 24.0 O = 24.0 Ne + 26.0 Al
Q =:  908.310045834  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 P + 24.0 F = 24.0 Na + 26.0 Al
Q =:  908.310045834  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 P + 24.0 Ne = 24.0 Mg + 26.0 Al
Q =:  908.310045834  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 P + 24.0 Na = 24.0 Al + 26.0 Al
Q =:  908.310045834  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 P + 24.0 Mg = 24.0 Si + 26.0 Al
Q =:  908.310045834  MeV
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Reacción: 26.0 P + 24.0 Al = 24.0 P + 26.0 Al
Q =:  908.310045834  MeV
~~~~~~~~~~~~~~~~~~~~~~

In [26]:
#El indexado se guardo en INDEX de las reacciones posibles
INDEX

[[154, 2, 1],
 [155, 1, 1],
 [155, 2, 2],
 [156, 1, 2],
 [155, 3, 3],
 [153, 6, 4],
 [154, 5, 4],
 [154, 6, 5],
 [155, 4, 4],
 [155, 5, 5],
 [155, 6, 6],
 [156, 4, 5],
 [156, 5, 6],
 [157, 4, 6],
 [153, 9, 7],
 [154, 8, 7],
 [154, 9, 8],
 [155, 7, 7],
 [155, 8, 8],
 [155, 9, 9],
 [156, 7, 8],
 [156, 8, 9],
 [157, 7, 9],
 [152, 13, 10],
 [153, 12, 10],
 [153, 13, 11],
 [154, 11, 10],
 [154, 12, 11],
 [154, 13, 12],
 [155, 10, 10],
 [155, 11, 11],
 [155, 12, 12],
 [155, 13, 13],
 [156, 10, 11],
 [156, 11, 12],
 [156, 12, 13],
 [157, 10, 12],
 [157, 11, 13],
 [158, 10, 13],
 [151, 18, 14],
 [152, 17, 14],
 [152, 18, 15],
 [153, 16, 14],
 [153, 17, 15],
 [153, 18, 16],
 [154, 15, 14],
 [154, 16, 15],
 [154, 17, 16],
 [154, 18, 17],
 [155, 14, 14],
 [155, 15, 15],
 [155, 16, 16],
 [155, 17, 17],
 [155, 18, 18],
 [156, 14, 15],
 [156, 15, 16],
 [156, 16, 17],
 [156, 17, 18],
 [157, 14, 16],
 [157, 15, 17],
 [157, 16, 18],
 [158, 14, 17],
 [158, 15, 18],
 [151, 23, 19],
 [152, 22, 19],
 [152,

Para calcular todas las posibilidades existentes de ${^{26}_{13}Al}$ sólo hay que cambiar $An$ por 295

```python
F F M G
```